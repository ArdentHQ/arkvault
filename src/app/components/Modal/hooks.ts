import { useCallback, useEffect, useMemo, useState } from "react";
import { browser } from "@/utils/platform";

export const observeElementHeight = (modalElement: HTMLDivElement, onResize: () => undefined) => {
	const observer = new ResizeObserver(() => onResize());

	observer.observe(modalElement);
	return observer;
};

const useModalOverflowYOffset = ({
	modalContainerReference,
}: {
	modalContainerReference: { current: HTMLDivElement | null };
}) => {
	const [overflowYClass, setOffsetYClass] = useState("overflow-y-hidden");

	useEffect(() => {
		const updateModalOffset = () => {
			const windowHeight = window.innerHeight;
			const currentModalHeight = modalContainerReference?.current?.clientHeight || 0;

			if (windowHeight - currentModalHeight < 85) {
				setOffsetYClass("overflow-y-auto");
			} else {
				setOffsetYClass("overflow-y-hidden");
			}

			return undefined;
		};

		let observer: ResizeObserver;
		if (!!modalContainerReference?.current) {
			observer = observeElementHeight(modalContainerReference.current, updateModalOffset);
		}

		window.addEventListener("resize", updateModalOffset);
		updateModalOffset();

		return () => {
			window.removeEventListener("resize", updateModalOffset);
			observer?.disconnect();
		};
	}, [modalContainerReference.current]);

	return {
		overflowYClass,
	};
};

export const useModal = ({
	isOpen,
	onClose,
	modalContainerReference,
}: {
	isOpen: boolean;
	onClose?: any;
	modalContainerReference: { current: HTMLDivElement | null };
}) => {
	const defaultOverflow = useMemo(() => (browser.supportsOverflowOverlay() ? "overlay" : "scroll"), []);
	const { overflowYClass } = useModalOverflowYOffset({ modalContainerReference });

	useEffect(() => {
		document.body.style.overflowY = defaultOverflow;

		if (isOpen) {
			document.body.style.overflowY = "hidden";
		}

		return () => {
			document.body.style.overflowY = defaultOverflow;
		};
	}, [defaultOverflow, isOpen]);

	const onEscKey = useCallback(
		(event: KeyboardEvent) => {
			event.preventDefault();
			event.stopPropagation();
			if (event.key === "Escape") {
				onClose();
			}
		},
		[onClose],
	);

	useEffect(() => {
		if (isOpen) {
			document.addEventListener("keyup", onEscKey, false);
		} else {
			document.removeEventListener("keyup", onEscKey);
		}

		return () => {
			document.removeEventListener("keyup", onEscKey);
		};
	}, [isOpen, onEscKey]);

	return {
		overflowYClass,
	};
};
