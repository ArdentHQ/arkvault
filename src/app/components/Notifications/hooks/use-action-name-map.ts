import { useTranslation } from "react-i18next";

import { Action } from "@/app/components/Notifications/Notifications.contracts";

export const useActionNameMap = () => {
	const { t } = useTranslation();

	const mapActionName = (actionName: string) => {
		const actionsMap: { [key: string]: Action } = {
			"Read Changelog": {
				label: t("COMMON.NOTIFICATIONS.ACTIONS.READ_CHANGELOG"),
				value: "changelog",
			},
			update: {
				label: t("COMMON.NOTIFICATIONS.ACTIONS.UPDATE"),
				value: "update",
			},
		};
		return actionsMap[actionName] || { label: actionName, value: actionName };
	};

	return { mapActionName };
};
