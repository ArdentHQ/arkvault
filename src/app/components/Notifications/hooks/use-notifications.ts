import { Contracts } from "@payvo/sdk-profiles";
import { useMemo } from "react";

export const useNotifications = ({ profile }: { profile: Contracts.IProfile }) => {
	const isSyncing = profile.notifications().transactions().isSyncing();

	const { markAllTransactionsAsRead, markAsRead, releases, transactions } = useMemo(() => {
		const markAllTransactionsAsRead = (isVisible: boolean) => {
			if (!isVisible) {
				return;
			}

			profile.notifications().transactions().markAllAsRead();
		};

		const markAsRead = (isVisible: boolean, id: string) => {
			if (!isVisible) {
				return;
			}

			profile.notifications().markAsRead(id);
		};

		return {
			markAllTransactionsAsRead,
			markAsRead,
			releases: profile.notifications().releases().recent(),
			transactions: profile.notifications().transactions().transactions(),
		};
	}, [profile, isSyncing]); // eslint-disable-line react-hooks/exhaustive-deps

	return {
		hasUnread: (releases.length > 0 || transactions.length > 0) && profile.notifications().hasUnread(),
		markAllTransactionsAsRead,
		markAsRead,
		releases,
		transactions,
	};
};
