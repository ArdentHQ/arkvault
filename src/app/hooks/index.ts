export * from "./click-outside";
export * from "./debounce";
export * from "./env";
export * from "./network-status";
export * from "./use-accent-color";
export * from "./use-clipboard";
export * from "./use-deeplink";
export * from "./use-document-title";
export * from "./use-fees";
export * from "./use-ledger-model-status";
export * from "./use-locale-currency";
export * from "./use-network-options";
export * from "./use-password-validation";
export * from "./use-previous";
export * from "./use-profile-synchronizer";
export * from "./use-query-parameters";
export * from "./use-random-number";
export * from "./use-reload-path";
export * from "./use-scroll";
export * from "./use-synchronizer";
export * from "./use-theme";
export * from "./use-validation";
export * from "./use-wallet-alias";
export * from "./use-breakpoint";
export * from "./use-accordion";
export * from "./use-server-health-status";
