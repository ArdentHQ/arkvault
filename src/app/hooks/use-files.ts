import { fileOpen, fileSave } from "browser-fs-access";

interface BaseDialogOptions {
	extensions?: string[];
	description?: string;
}

type OpenFileDialogOptions = {
	mimeTypes?: string[];
} & BaseDialogOptions;

type SaveFileDialogOptions = {
	fileName?: string;
} & BaseDialogOptions;

interface ReadableFile {
	content: string;
	extension: string;
	name: string;
}

interface ValidateOptions {
	extensions?: string[];
}

const isLegacy = () => !("showSaveFilePicker" in window);

const getExtension = (fileName?: string) => fileName?.split(".").pop() || "";

const waitFileRead = (file: File, resolve: (file?: ReadableFile) => void) => {
	const reader = new FileReader();

	reader.addEventListener("load", (event) => {
		if (!event?.target?.result) {
			return resolve(undefined);
		}

		resolve({
			content: event.target.result as string,
			extension: getExtension(file.name),
			name: file?.name,
		});
	});

	return reader;
};

const readFileAsDataUri = async (file: File): Promise<ReadableFile | undefined> =>
	new Promise((resolve) => waitFileRead(file, resolve).readAsDataURL(file));

const readFileAsText = async (file: File): Promise<ReadableFile | undefined> =>
	new Promise((resolve) => waitFileRead(file, resolve).readAsText(file));

const sanitizeExtensions = (extensions?: string[]) => {
	if (extensions) {
		const sanitized: string[] = [];

		for (const extension of extensions) {
			sanitized.push(extension.startsWith(".") ? extension : `.${extension}`);
		}

		return sanitized;
	}
};

const showOpenDialog = async (options: OpenFileDialogOptions): Promise<File | undefined> => {
	if (options.extensions) {
		options.extensions = sanitizeExtensions(options.extensions);
	}

	return fileOpen(options);
};

const showSaveDialog = async (contents: string, options: SaveFileDialogOptions): Promise<string | undefined> => {
	if (options.fileName && !options.extensions) {
		options.extensions = [getExtension(options.fileName)];
	}

	options.extensions = sanitizeExtensions(options.extensions);

	const result = await fileSave(new Blob([contents], { type: "data/text" }), options);

	return result?.name;
};

const showImageSaveDialog = async (contents: string, options: SaveFileDialogOptions): Promise<string | undefined> => {
	if (options.fileName && !options.extensions) {
		options.extensions = [getExtension(options.fileName)];
	}

	options.extensions = sanitizeExtensions(options.extensions);

	const file = await fetch(contents);

	const result = await fileSave(await file.blob(), options);

	return result?.name;
};

const isValidImage = async (file: ReadableFile, options?: ValidateOptions): Promise<boolean> => {
	if (!file.extension || (options?.extensions && !options?.extensions?.includes(file.extension))) {
		return false;
	}

	if (!file.content.startsWith("data:image")) {
		return false;
	}

	const image = new Image();
	image.src = file.content;

	try {
		await image.decode();
	} catch {
		return false;
	}

	return true;
};

const openImage = async (options?: ValidateOptions): Promise<ReadableFile> => {
	const raw = await showOpenDialog({ extensions: options?.extensions });
	const file = await readFileAsDataUri(raw!);

	if (!file) {
		throw new Error("Empty");
	}

	const isValid = await isValidImage(file, options);

	if (!isValid) {
		throw new Error("Invalid image");
	}

	return file;
};

interface UseFilesOutput {
	getExtension: (fileName?: string) => string;
	isLegacy: () => boolean;
	readFileAsDataUri: (file: File) => Promise<ReadableFile | undefined>;
	readFileAsText: (file: File) => Promise<ReadableFile | undefined>;
	showOpenDialog: (options: OpenFileDialogOptions) => Promise<File | undefined>;
	showSaveDialog: (contents: string, options: SaveFileDialogOptions) => Promise<string | undefined>;
	showImageSaveDialog: (contents: string, options: SaveFileDialogOptions) => Promise<string | undefined>;
	openImage: (options?: ValidateOptions) => Promise<ReadableFile>;
}

const useFiles = (): UseFilesOutput => ({
	getExtension,
	isLegacy,
	openImage,
	readFileAsDataUri,
	readFileAsText,
	showImageSaveDialog,
	showOpenDialog,
	showSaveDialog,
});

export { isValidImage, useFiles };

export type { ReadableFile };
