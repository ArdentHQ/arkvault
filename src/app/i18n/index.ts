import i18n from "i18next";
import { initReactI18next } from "react-i18next";

// Helpers
import { buildTranslations } from "./helpers";

const defaultNS = "translation";

const resources = {
	en: {
		translation: buildTranslations(),
	},
};

i18n.use(initReactI18next).init({
	defaultNS,
	lng: "en",
	ns: [defaultNS],
	resources,
});

export { defaultNS, resources };

export { default as i18n } from "i18next";
