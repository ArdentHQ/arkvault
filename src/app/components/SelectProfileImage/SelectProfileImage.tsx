import cn from "classnames";
import React, { useMemo } from "react";
import { useTranslation } from "react-i18next";
import tw, { styled } from "twin.macro";

import { Button } from "@/app/components/Button";
import { Icon } from "@/app/components/Icon";
import { useFiles } from "@/app/hooks/use-files";
import { toasts } from "@/app/services";

interface SelectProfileImageProperties {
	className?: string;
	value?: string;
	name?: string;
	showLabel?: boolean;
	onSelect: (raw: string) => void;
}

const UploadButtonWrapper = styled.div`
	${tw`h-full w-full`}

	button {
		${tw`h-full w-full`}
		${tw`focus:ring-0!`}

		&:not(:focus):hover:enabled {
			${tw`bg-theme-secondary-900 dark:bg-theme-secondary-600 opacity-85`};
		}
	}
`;

const ProfileImageStyled = styled.div`
	& {
		${tw`relative inline-flex items-center justify-center rounded-md overflow-hidden cursor-pointer h-full`};
		${tw`focus-within:(ring-2 ring-theme-primary-400)`};
	}

	&:after {
		content: "";
		box-shadow: 0 0 0 25px rgba(0, 0, 0, 0.4);
		${tw`absolute inset-1 rounded-full`};
	}

	&:hover .upload-button-overlay {
		${tw`opacity-100`};
	}
`;

const ALLOWED_EXTENSIONS = ["png", "jpg", "jpeg", "bmp"];

export const SelectProfileImage = ({
	className,
	value,
	name,
	showLabel = true,
	onSelect,
}: SelectProfileImageProperties) => {
	const { t } = useTranslation();
	const { openImage } = useFiles();

	const handleOpenFile = async () => {
		try {
			const image = await openImage({ extensions: ALLOWED_EXTENSIONS });

			onSelect(image?.content);
		} catch (error) {
			if (!error.message.includes("The user aborted a request")) {
				toasts.error(t("COMMON.ERRORS.INVALID_IMAGE"));
			}
		}
	};

	const isSvg = useMemo(() => value?.endsWith("</svg>"), [value]);

	const renderButton = () => {
		if (value) {
			return (
				<div className="relative z-0 h-20 w-20">
					<ProfileImageStyled>
						<img
							data-testid={`SelectProfileImage__avatar-${isSvg ? "identicon" : "image"}`}
							src={isSvg ? `data:image/svg+xml;utf8,${value}` : value}
							className="min-h-full object-cover"
							alt="Avatar"
						/>

						{isSvg && (
							<span className="absolute text-2xl font-semibold text-white">
								{name?.slice(0, 2).toUpperCase()}
							</span>
						)}

						<button
							type="button"
							className="upload-button-overlay absolute z-10 h-full w-full overflow-hidden p-1 opacity-0 transition-opacity duration-200 focus:outline-none"
							onClick={handleOpenFile}
							data-testid="SelectProfileImage__upload-button"
						>
							<div className="flex h-full items-center justify-center rounded-full bg-theme-secondary-900 opacity-85 dark:bg-black">
								<Icon
									name="ArrowUpBracket"
									className="text-white dark:text-theme-secondary-200"
									size="lg"
								/>
							</div>
						</button>
					</ProfileImageStyled>

					{!isSvg && (
						<div className="absolute -top-2 -right-2 z-20">
							<Button
								size="icon"
								variant="danger"
								className="flex h-5 w-5 items-center justify-center p-0"
								onClick={() => onSelect("")}
								data-testid="SelectProfileImage__remove-button"
							>
								<Icon name="Cross" size="sm" />
							</Button>
						</div>
					)}
				</div>
			);
		}

		return (
			<div className="h-20 w-20 rounded-md border-2 border-dashed border-theme-secondary-400 p-1 focus-within:border-solid focus-within:border-theme-primary-400 dark:border-theme-secondary-700">
				<div className="h-full overflow-hidden rounded-full">
					<UploadButtonWrapper>
						<Button
							variant="secondary"
							onClick={handleOpenFile}
							data-testid="SelectProfileImage__upload-button"
						>
							<Icon
								name="ArrowUpBracket"
								className="text-theme-primary-600 dark:text-theme-secondary-200"
								size="lg"
							/>
						</Button>
					</UploadButtonWrapper>
				</div>
			</div>
		);
	};

	return (
		<div className={cn("group space-y-2", className)}>
			{showLabel && (
				<span className="cursor-default text-sm font-semibold text-theme-secondary-text transition-colors duration-100 group-hover:text-theme-primary-600">
					{t("SETTINGS.GENERAL.PERSONAL.PROFILE_IMAGE")}
				</span>
			)}

			<div className="flex flex-row">{renderButton()}</div>
		</div>
	);
};
