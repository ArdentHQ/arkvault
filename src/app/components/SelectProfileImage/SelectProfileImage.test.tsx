import { renderHook } from "@testing-library/react-hooks";
import userEvent from "@testing-library/user-event";
import React from "react";
import * as browserAccess from "browser-fs-access";

import { SelectProfileImage } from "./SelectProfileImage";
import { useFiles } from "@/app/hooks/use-files";
import { translations } from "@/app/i18n/common/i18n";
import { toasts } from "@/app/services";
import { render, screen, waitFor, MockFile } from "@/utils/testing-library";

const uploadButton = () => screen.getByTestId("SelectProfileImage__upload-button");

describe("SelectProfileImage", () => {
	it("should render", () => {
		const onSelect = vi.fn();

		const { asFragment } = render(<SelectProfileImage onSelect={onSelect} />);

		expect(asFragment()).toMatchSnapshot();
	});

	it("should render with value svg", () => {
		const onSelect = vi.fn();

		const { asFragment } = render(<SelectProfileImage value="<svg>test</svg>" onSelect={onSelect} />);

		expect(asFragment()).toMatchSnapshot();
	});

	it("should render without value svg", () => {
		const onSelect = vi.fn();

		const { asFragment } = render(<SelectProfileImage value="test" onSelect={onSelect} />);

		expect(asFragment()).toMatchSnapshot();

		userEvent.click(screen.getByTestId("SelectProfileImage__remove-button"));

		expect(onSelect).toHaveBeenCalledWith(expect.any(String));
	});

	it("should handle upload file", async () => {
		const browserAccessMock = vi
			.spyOn(browserAccess, "fileOpen")
			.mockResolvedValue(new MockFile([""], { name: "test.png", type: "image/png" }));

		const onSelect = vi.fn();

		const { asFragment } = render(<SelectProfileImage value="test" onSelect={onSelect} />);
		const { result: useFilesResult } = renderHook(() => useFiles());

		expect(asFragment()).toMatchSnapshot();

		uploadButton().addEventListener("click", browserAccessMock as any);
		userEvent.click(uploadButton());

		await expect(
			useFilesResult.current.showOpenDialog({
				extensions: [".png", ".jpg", ".jpeg", ".bmp"],
			}),
		).resolves.not.toThrow();

		await waitFor(() => expect(onSelect).toHaveBeenCalledWith("data:image/png;base64,"));

		browserAccessMock.mockRestore();
	});

	it("should not allow to upload an invalid file image", async () => {
		const onSelect = vi.fn();
		const toastSpy = vi.spyOn(toasts, "error");

		const { asFragment } = render(<SelectProfileImage value="test" onSelect={onSelect} />);

		expect(asFragment()).toMatchSnapshot();

		const browserAccessMock = vi
			.spyOn(browserAccess, "fileOpen")
			.mockResolvedValue(new File(["123"], "not-an-image.png"));

		uploadButton().addEventListener("click", browserAccessMock as any);
		userEvent.click(uploadButton());

		await waitFor(() => expect(toastSpy).toHaveBeenCalledWith(translations.ERRORS.INVALID_IMAGE));

		Object.defineProperty(global.FileReader.prototype, "readAsDataURL", {
			value: () => {
				throw new Error("Wrong file");
			},
		});

		userEvent.click(uploadButton());

		await waitFor(() => expect(toastSpy).toHaveBeenCalledWith(translations.ERRORS.INVALID_IMAGE));

		browserAccessMock.mockRestore();
	});

	it("should not handle upload file", async () => {
		const onSelect = vi.fn();
		const toastSpy = vi.spyOn(toasts, "error");

		const { asFragment } = render(<SelectProfileImage value="test" onSelect={onSelect} />);

		expect(asFragment()).toMatchSnapshot();

		const browserAccessMock = vi.spyOn(browserAccess, "fileOpen").mockResolvedValue(undefined);

		userEvent.click(uploadButton());

		await waitFor(() => expect(toastSpy).toHaveBeenCalledWith(translations.ERRORS.INVALID_IMAGE));
		await waitFor(() => expect(onSelect).not.toHaveBeenCalled());

		browserAccessMock.mockRestore();
		toastSpy.mockRestore();
	});

	it("should not allow to upload when user aborted a request", async () => {
		const onSelect = vi.fn();
		const toastSpy = vi.spyOn(toasts, "error");

		const { asFragment } = render(<SelectProfileImage value="test" onSelect={onSelect} />);

		expect(asFragment()).toMatchSnapshot();

		const browserAccessMock = vi
			.spyOn(browserAccess, "fileOpen")
			.mockRejectedValue(new Error("The user aborted a request."));

		userEvent.click(uploadButton());

		await waitFor(() => expect(toastSpy).not.toHaveBeenCalled());

		browserAccessMock.mockRestore();
		toastSpy.mockRestore();
	});
});
