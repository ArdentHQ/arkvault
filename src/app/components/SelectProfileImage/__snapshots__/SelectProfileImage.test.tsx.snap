// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectProfileImage should handle upload file 1`] = `
<DocumentFragment>
  <div
    class="group space-y-2"
  >
    <span
      class="cursor-default text-sm font-semibold text-theme-secondary-text transition-colors duration-100 group-hover:text-theme-primary-600"
    >
      Profile Image
    </span>
    <div
      class="flex flex-row"
    >
      <div
        class="relative z-0 h-20 w-20"
      >
        <div
          class="css-ikig5u"
        >
          <img
            alt="Avatar"
            class="min-h-full object-cover"
            data-testid="SelectProfileImage__avatar-image"
            src="test"
          />
          <button
            class="upload-button-overlay absolute z-10 h-full w-full overflow-hidden p-1 opacity-0 transition-opacity duration-200 focus:outline-none"
            data-testid="SelectProfileImage__upload-button"
            type="button"
          >
            <div
              class="flex h-full items-center justify-center rounded-full bg-theme-secondary-900 opacity-85 dark:bg-black"
            >
              <div
                class="text-white dark:text-theme-secondary-200 css-1i4b0eq"
                height="20"
                width="20"
              >
                <svg>
                  arrow-up-bracket.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
        <div
          class="absolute -top-2 -right-2 z-20"
        >
          <button
            class="flex h-5 w-5 items-center justify-center p-0 css-1km99ls"
            data-testid="SelectProfileImage__remove-button"
            type="button"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                class="css-15txs7d"
                height="10"
                width="10"
              >
                <svg>
                  cross.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectProfileImage should not allow to upload an invalid file image 1`] = `
<DocumentFragment>
  <div
    class="group space-y-2"
  >
    <span
      class="cursor-default text-sm font-semibold text-theme-secondary-text transition-colors duration-100 group-hover:text-theme-primary-600"
    >
      Profile Image
    </span>
    <div
      class="flex flex-row"
    >
      <div
        class="relative z-0 h-20 w-20"
      >
        <div
          class="css-ikig5u"
        >
          <img
            alt="Avatar"
            class="min-h-full object-cover"
            data-testid="SelectProfileImage__avatar-image"
            src="test"
          />
          <button
            class="upload-button-overlay absolute z-10 h-full w-full overflow-hidden p-1 opacity-0 transition-opacity duration-200 focus:outline-none"
            data-testid="SelectProfileImage__upload-button"
            type="button"
          >
            <div
              class="flex h-full items-center justify-center rounded-full bg-theme-secondary-900 opacity-85 dark:bg-black"
            >
              <div
                class="text-white dark:text-theme-secondary-200 css-1i4b0eq"
                height="20"
                width="20"
              >
                <svg>
                  arrow-up-bracket.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
        <div
          class="absolute -top-2 -right-2 z-20"
        >
          <button
            class="flex h-5 w-5 items-center justify-center p-0 css-1km99ls"
            data-testid="SelectProfileImage__remove-button"
            type="button"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                class="css-15txs7d"
                height="10"
                width="10"
              >
                <svg>
                  cross.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectProfileImage should not allow to upload when user aborted a request 1`] = `
<DocumentFragment>
  <div
    class="group space-y-2"
  >
    <span
      class="cursor-default text-sm font-semibold text-theme-secondary-text transition-colors duration-100 group-hover:text-theme-primary-600"
    >
      Profile Image
    </span>
    <div
      class="flex flex-row"
    >
      <div
        class="relative z-0 h-20 w-20"
      >
        <div
          class="css-ikig5u"
        >
          <img
            alt="Avatar"
            class="min-h-full object-cover"
            data-testid="SelectProfileImage__avatar-image"
            src="test"
          />
          <button
            class="upload-button-overlay absolute z-10 h-full w-full overflow-hidden p-1 opacity-0 transition-opacity duration-200 focus:outline-none"
            data-testid="SelectProfileImage__upload-button"
            type="button"
          >
            <div
              class="flex h-full items-center justify-center rounded-full bg-theme-secondary-900 opacity-85 dark:bg-black"
            >
              <div
                class="text-white dark:text-theme-secondary-200 css-1i4b0eq"
                height="20"
                width="20"
              >
                <svg>
                  arrow-up-bracket.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
        <div
          class="absolute -top-2 -right-2 z-20"
        >
          <button
            class="flex h-5 w-5 items-center justify-center p-0 css-1km99ls"
            data-testid="SelectProfileImage__remove-button"
            type="button"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                class="css-15txs7d"
                height="10"
                width="10"
              >
                <svg>
                  cross.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectProfileImage should not handle upload file 1`] = `
<DocumentFragment>
  <div
    class="group space-y-2"
  >
    <span
      class="cursor-default text-sm font-semibold text-theme-secondary-text transition-colors duration-100 group-hover:text-theme-primary-600"
    >
      Profile Image
    </span>
    <div
      class="flex flex-row"
    >
      <div
        class="relative z-0 h-20 w-20"
      >
        <div
          class="css-ikig5u"
        >
          <img
            alt="Avatar"
            class="min-h-full object-cover"
            data-testid="SelectProfileImage__avatar-image"
            src="test"
          />
          <button
            class="upload-button-overlay absolute z-10 h-full w-full overflow-hidden p-1 opacity-0 transition-opacity duration-200 focus:outline-none"
            data-testid="SelectProfileImage__upload-button"
            type="button"
          >
            <div
              class="flex h-full items-center justify-center rounded-full bg-theme-secondary-900 opacity-85 dark:bg-black"
            >
              <div
                class="text-white dark:text-theme-secondary-200 css-1i4b0eq"
                height="20"
                width="20"
              >
                <svg>
                  arrow-up-bracket.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
        <div
          class="absolute -top-2 -right-2 z-20"
        >
          <button
            class="flex h-5 w-5 items-center justify-center p-0 css-1km99ls"
            data-testid="SelectProfileImage__remove-button"
            type="button"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                class="css-15txs7d"
                height="10"
                width="10"
              >
                <svg>
                  cross.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectProfileImage should render 1`] = `
<DocumentFragment>
  <div
    class="group space-y-2"
  >
    <span
      class="cursor-default text-sm font-semibold text-theme-secondary-text transition-colors duration-100 group-hover:text-theme-primary-600"
    >
      Profile Image
    </span>
    <div
      class="flex flex-row"
    >
      <div
        class="h-20 w-20 rounded-md border-2 border-dashed border-theme-secondary-400 p-1 focus-within:border-solid focus-within:border-theme-primary-400 dark:border-theme-secondary-700"
      >
        <div
          class="h-full overflow-hidden rounded-full"
        >
          <div
            class="css-5aibro"
          >
            <button
              class="css-1jhy6v9"
              data-testid="SelectProfileImage__upload-button"
              type="button"
            >
              <div
                class="flex items-center space-x-2"
              >
                <div
                  class="text-theme-primary-600 dark:text-theme-secondary-200 css-1i4b0eq"
                  height="20"
                  width="20"
                >
                  <svg>
                    arrow-up-bracket.svg
                  </svg>
                </div>
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectProfileImage should render with value svg 1`] = `
<DocumentFragment>
  <div
    class="group space-y-2"
  >
    <span
      class="cursor-default text-sm font-semibold text-theme-secondary-text transition-colors duration-100 group-hover:text-theme-primary-600"
    >
      Profile Image
    </span>
    <div
      class="flex flex-row"
    >
      <div
        class="relative z-0 h-20 w-20"
      >
        <div
          class="css-ikig5u"
        >
          <img
            alt="Avatar"
            class="min-h-full object-cover"
            data-testid="SelectProfileImage__avatar-identicon"
            src="data:image/svg+xml;utf8,<svg>test</svg>"
          />
          <span
            class="absolute text-2xl font-semibold text-white"
          />
          <button
            class="upload-button-overlay absolute z-10 h-full w-full overflow-hidden p-1 opacity-0 transition-opacity duration-200 focus:outline-none"
            data-testid="SelectProfileImage__upload-button"
            type="button"
          >
            <div
              class="flex h-full items-center justify-center rounded-full bg-theme-secondary-900 opacity-85 dark:bg-black"
            >
              <div
                class="text-white dark:text-theme-secondary-200 css-1i4b0eq"
                height="20"
                width="20"
              >
                <svg>
                  arrow-up-bracket.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectProfileImage should render without value svg 1`] = `
<DocumentFragment>
  <div
    class="group space-y-2"
  >
    <span
      class="cursor-default text-sm font-semibold text-theme-secondary-text transition-colors duration-100 group-hover:text-theme-primary-600"
    >
      Profile Image
    </span>
    <div
      class="flex flex-row"
    >
      <div
        class="relative z-0 h-20 w-20"
      >
        <div
          class="css-ikig5u"
        >
          <img
            alt="Avatar"
            class="min-h-full object-cover"
            data-testid="SelectProfileImage__avatar-image"
            src="test"
          />
          <button
            class="upload-button-overlay absolute z-10 h-full w-full overflow-hidden p-1 opacity-0 transition-opacity duration-200 focus:outline-none"
            data-testid="SelectProfileImage__upload-button"
            type="button"
          >
            <div
              class="flex h-full items-center justify-center rounded-full bg-theme-secondary-900 opacity-85 dark:bg-black"
            >
              <div
                class="text-white dark:text-theme-secondary-200 css-1i4b0eq"
                height="20"
                width="20"
              >
                <svg>
                  arrow-up-bracket.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
        <div
          class="absolute -top-2 -right-2 z-20"
        >
          <button
            class="flex h-5 w-5 items-center justify-center p-0 css-1km99ls"
            data-testid="SelectProfileImage__remove-button"
            type="button"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                class="css-15txs7d"
                height="10"
                width="10"
              >
                <svg>
                  cross.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
